Bitset for edges for BC-level

l-hoang opened the pull request:
Adds a bitset to mark active edges for bc-level. What this does is make
it so the backward Brandes step does not have to read the edge
destination as well as the distance label on the destination to
determine if the edge is part of the successors list of that node.

ddn0 commented:
Now that I added date to GaloisConfig.cmake, conda is less happy about howardhinnant/date just being a host dependency and not a run dependency:

https://github.com/KatanaGraph/katana/pull/285/checks?check_run_id=924902024

I opted to just make it a run dependency too, but if there is an obvious (better?) way to indicate a compile-time dependency versus a runtime one, I'm all ears.

arthurp commented:
> Now that I added date to GaloisConfig.cmake, conda is less happy about howardhinnant/date just being a host dependency and not a run dependency:
> 
> https://github.com/KatanaGraph/katana/pull/285/checks?check_run_id=924902024
> 
> I opted to just make it a run dependency too, but if there is an obvious (better?) way to indicate a compile-time dependency versus a runtime one, I'm all ears.

Well I think it is a "runtime" dependency for this package. Runtime for a package means when it's installed and used by the user, so for libraries that will mean compile time for the application, so if date is mentioned in our installed headers than it needs to be a runtime dependency of the package.

arthurp commented:
I think you need to change the pyarrow dependency in `python/conda_recipe/meta.yaml` it also specifies version numbers. Though maybe I could get away without since the version of arrow-cpp will be set by galois, which will force the version of arrow-cpp.

BTW, @insertinterestingnamehere it looks like we have a version pinning issue here. The package installed correctly with arrow-cpp=0.17 even though it was built with arrow-cpp=1.0 and then it failed to work. So we do need to figure out that version pinning issue at some point.

arthurp commented:
It looks like pyarrow 1.0 returns special types that are not compatible with normal python types in more cases. I hate that quirk of that library. Thanks to unit testing I know about the problem. I think the fix for now is just to unwrap the pyarrow types in the unit test. Shall I push a change to the PR? Or do you want to do it @ddn0?

ddn0 commented:
> Shall I push a change to the PR? Or do you want to do it @ddn0?

If you know the answer, you can take over :D

ddn0 commented:
@arthurp qq: does `as_py` also work in pyarrow 0.17? I don't think we need to support both versions, but I'm just curious.

arthurp commented:
Yes. The tests pass on both 0.17 and 1.0. There are some things added in 1.0 though, so I expect in time we will drift towards 1.0 dependency.

Commit:
Bitset for edges for BC-level

Adds a bitset to mark active edges for bc-level. What this does is make
it so the backward Brandes step does not have to read the edge
destination as well as the distance label on the destination to
determine if the edge is part of the successors list of that node.

roshandathathri reviewed (approved):
LGTM. Thanks!
